api/externalLinks/ExternalLinksRestHandler.go

```diff
@@ -0,0 +1,218 @@
+	"github.com/devtron-labs/devtron/pkg/user"
+	"github.com/devtron-labs/devtron/pkg/user/casbin"
+	"github.com/gorilla/mux"
+	"github.com/juju/errors"
+	"go.uber.org/zap"
+	"gopkg.in/go-playground/validator.v9"
+	"net/http"
+	"strconv"
+)
+
+type ExternalLinksRestHandler interface {
+	CreateExternalLinks(w http.ResponseWriter, r *http.Request)
+	GetExternalLinksTools(w http.ResponseWriter, r *http.Request)
+	GetExternalLinks(w http.ResponseWriter, r *http.Request)
+	UpdateExternalLinks(w http.ResponseWriter, r *http.Request)
+	DeleteExternalLinks(w http.ResponseWriter, r *http.Request) // Update is_active to false link
+}
+type ExternalLinksRestHandlerImpl struct {
+	logger               *zap.SugaredLogger
+	externalLinksService externalLinks.ExternalLinksService
+	userService          user.UserService
+	validator            *validator.Validate
+	enforcer             casbin.Enforcer
+	userAuthService      user.UserAuthService
+	deleteService        delete2.DeleteService
+}
+
+func NewExternalLinksRestHandlerImpl(logger *zap.SugaredLogger,
+	externalLinksService externalLinks.ExternalLinksService,
+	userService user.UserService,
+	enforcer casbin.Enforcer,
+	validator *validator.Validate, userAuthService user.UserAuthService,
+	deleteService delete2.DeleteService,
+) *ExternalLinksRestHandlerImpl {
+	return &ExternalLinksRestHandlerImpl{
+		logger:               logger,
+		externalLinksService: externalLinksService,
+		userService:          userService,
+		validator:            validator,
+		enforcer:             enforcer,
+		userAuthService:      userAuthService,
+		deleteService:        deleteService,
+	}
+}
+func (impl ExternalLinksRestHandlerImpl) CreateExternalLinks(w http.ResponseWriter, r *http.Request) {
+	userId, err := impl.userService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+	decoder := json.NewDecoder(r.Body)
+	var beans []*externalLinks.ExternalLinksRequest
+	err = decoder.Decode(&beans)
+	if err != nil {
+		impl.logger.Errorw("request err, SaveLink", "err", err, "payload", beans)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+
+	token := r.Header.Get("token")
+	if ok := impl.enforcer.Enforce(token, casbin.ResourceGlobal, casbin.ActionCreate, "*"); !ok {
+		common.WriteJsonResp(w, errors.New("unauthorized"), nil, http.StatusForbidden)
+		return
+	}
+
+	res, err := impl.externalLinksService.Create(beans, userId)
+	if err != nil {
+		impl.logger.Errorw("service err, SaveLink", "err", err, "payload", beans)
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+	common.WriteJsonResp(w, err, res, http.StatusOK)
+}
+func (impl ExternalLinksRestHandlerImpl) GetExternalLinksTools(w http.ResponseWriter, r *http.Request) {
+	userId, err := impl.userService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
```

Comment:
if userId is 0 then error should be "UnAuthenticated" not "UnAuthorized". (However I can see this thing everywhere. so we can do this. later)